src/main/java/uk/gov/hmcts/reform/iacaseapi/infrastructure/clients/ClientResponseErrorHandler.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmcts.reform.iacaseapi.infrastructure.clients;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.stream.Collectors;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.web.client.DefaultResponseErrorHandler;
+
+public class ClientResponseErrorHandler extends DefaultResponseErrorHandler  {
+
+    @Override
+    public void handleError(ClientHttpResponse response) throws IOException {
+
+        if (response.getStatusCode().is4xxClientError() || response.getStatusCode().is5xxServerError()) {
+
+            try(BufferedReader reader = new BufferedReader(new InputStreamReader(response.getBody()))) {
```

Comment:
please check if response.getBody() is not null

Reply:
Null check added